docker/tarfile/dest.go

```diff
@@ -29,6 +29,7 @@ type Destination struct {
 	// Other state.
 	blobs  map[digest.Digest]types.BlobInfo // list of already-sent blobs
 	config []byte
+	sysCtx *types.SystemContext
```

Comment:
Nothing initializes this; this needs to be set in a new `NewDestinationWithContext` (and callers updated)